crates/bevy_ecs/src/event.rs

```diff
@@ -813,33 +856,23 @@ pub fn signal_event_update_system(signal: Option<ResMut<EventUpdateSignal>>) {
     }
 }
 
-/// Resets the `EventUpdateSignal`
-pub fn reset_event_update_signal_system(signal: Option<ResMut<EventUpdateSignal>>) {
-    if let Some(mut s) = signal {
-        s.0 = false;
+/// A system that calls [`Events::update`] on all registered [`Events`] in the world.
+pub fn event_update_system(world: &mut World) {
+    if world.contains_resource::<EventRegistry>() {
```

Comment:
Is this better than just `init_resource` on the registry here?

Reply:
If for whatever reason, someone is running without any events in their World, this would avoid stealth allocating it.